lib/SILOptimizer/IPO/DeadFunctionElimination.cpp

```diff
@@ -468,6 +468,30 @@ class ExternalFunctionDefinitionsElimination : FunctionLivenessComputation {
     /// Therefore there is no need for a liveness computation.
     /// The next line can be just replaced by:
     /// return false;
+
+    // Keep all transparent functions alive. This is important because we have
+    // to generate code for transparent functions.
```

Comment:
For transparent functions no code is generated in the stdlib. Usually they are inlined and everything is fine. But if a function pointer is taken from a transparent function it can't be inlined and we need to generate code for it.
I'll describe that in the comment.